demos/gallery/samples/common/HTMLWidget.js

```diff
@@ -0,0 +1,62 @@
+import { HTMLWidget } from "@hpcc-js/common";
+
+const data = [
+    ["A", 100],
+    ["B", 162],
+    ["C", 262]
+];
+class HTMLExample extends HTMLWidget{
+    constructor(){
+        super();
+    }
+    enter(domNode, element){
+        super.enter(domNode, element);
+        this._div = element.append("div");
+    }
+    update(domNode, element){
+        super.update(domNode, element);
+        const gutter = 4;
+        const rowHeight = 30;
+        const labelWidth = 10;
+        
+        const rowsEnter = this._div.selectAll(".data-row").data(this.data())
+            .enter()
+            .append("div")
+            .style("width","100%")
+            .style("height",rowHeight+"px")
+            .style("margin-top",gutter+"px")
+            .classed("data-row", true)
+            ;
+        rowsEnter.append("span");
+        rowsEnter.append("div");
```

Comment:
I typicaly put these inside an "each" callback.

Reply:
IMO this implementation is more abstract and this is just sample code.